src/charm.py

```diff
@@ -372,6 +398,77 @@ def _is_magmad_enabled(self) -> bool:
         )
         return not magma_service.returncode
 
+    def _install_configurations(self, event: OrchestratorAvailableEvent) -> bool:
+        """Install or update configuration files.
+
+        Returns:
+            True if any changes were applied
+        """
+        return any(
+            [
+                self._install_root_ca_certificate(event.root_ca_certificate),
+                self._install_agw_config(event),
+            ]
+        )
+
+    @staticmethod
+    def _install_root_ca_certificate(root_ca_certificate: bytes) -> bool:
+        """Install root ca certificate or update it.
+
+        Returns:
+            True if the root ca certificate was written
+        """
+        cert_dir = Path(CERT_DIR)
+        if not cert_dir.exists():
+            cert_dir.mkdir()
+        root_ca_file = cert_dir / "rootCA.pem"
+        if root_ca_file.exists():
```

Comment:
How about replacing this bunch of `ifs` with a `try-except` block?

Reply:
I refactored the code and the `ifs` are a bit simpler, I think a `try-except` block would take more space and might be harder to follow, WDYT?

Reply:
Yeah, I definitely like that more :)